IF plugin registry form proposal (#2)
Green-Software-Foundation/if-plugin-website: .github/ISSUE_TEMPLATE/if-plugin-registry-form-template.md

---
name: IF Plugin Registry Form template
about: Use this template to submit an IF plugin for registry
title: "[PLUGIN REGISTRY]"
labels: ''
assignees: ''

---

## Plugin Name

> <!--- How do you call your plugin? -->

Reviewer: ```suggestion
<!--- What is the plugin name? -->
```

> ## Plugin GitHub URL

Reviewer: ```suggestion
## Link to Github/npm
```

> <!--- URL -->

Reviewer: ```suggestion
<!--- please provide the url for your plugin's Github repository and/or npm package page  -->
```

> ## Plugin Registry Basic Requirements

Reviewer: ```suggestion
## Plugin Registry Checklist
<!--- Please check off all the boxes that apply and provide requested links -->
```

> - [ ] My plugin executes a demo manifest yml with no errors or crashes.

Reviewer: ```suggestion
- [ ] My plugin integrates with IF, as demonstrated by execution from a manifest file with no errors or crashes.
```

> - [ ] My plugin has README file.

Reviewer: ```suggestion
- [ ] My plugin has README file.
  - LINK: //link to readme here
```

> - [ ] My plugin code includes a sample manifest .yaml file.

Reviewer: ```suggestion
- [ ] My plugin code includes a sample manifest .yaml file.
  - LINK //paste link to your sample manifest here
```

> - [ ] My plugin's code contain unit tests that cover >=80% of the plugins code.

Reviewer: I know I said that 100% coverage is a bit too strict, but I also think 80% is a bit arbitrary. The important branches might all be in the uncovered 20% and besides,t he threshold is arbitrary. Maybe we just ask for *some* unit tests and later we can reward projects that achieve 100% coverage with a badge.

- [ ] All unit tests that test my plugin are passing.

> - [ ] All other unit tests in the if-models repo are passing after my code is integrated into it.

Reviewer: ```suggestion
```

`if-models` is now `if-plugins`, and anyway the point of the plugin registry is that plugins won't make it into that repository - they will stay external. So I think this should just get deleted.

> - [ ] My plugin's code have no security vulnerabilities. 

Reviewer: ```suggestion
- [ ] I do not know of any security vulnerabilities associated with my code.
  <!-- link out to any relevant disclosures, disclaimers or supporting info as necessary --> 
```

> ## Define, in your own words, what is(are) the Requirement(s) of your plugin

Reviewer: ```suggestion
## What does your plugin do?
```

> <!--- I.E what is the basic functionality it must demonstrate. -->

Reviewer: ```suggestion
<!--- Explain in a sentence or two what the plugin does -->
```

> ## Please provide citations, links and references that support the validity of the implementation of your plugin

Reviewer: Let's actually deal with this by suggesting best practises in the registry docs instead of requiring it here

Author: I think we wanted to have this (and other answers) in this form so that:
1. Other ppl in the community can review any registered plugin at any time (transparency)
2. We can later review and revisit when we start randomly auditing plugins.
I don't mind citations, links and references being kept somewhere else, but I do think there should be only 1 place for all basic info for each register plugin. WDYT?

<!--- I.E provide reference to sources explaining how this implementation meets the requirement described above. -->

> ## Please describe the test cases covered by your plugin's dedicated unit tests

Reviewer: ```suggestion
```

Again, let's handle this elsewhere

Author: again, I don't mind. just think we need 1 place where all info (that is - the answers to the questions in this form) for all registered plugins reside in.
